eclair-core/src/main/scala/fr/acinq/eclair/tor/SttpUtils.scala

```diff
@@ -0,0 +1,49 @@
+package fr.acinq.eclair.tor
+
+import com.softwaremill.sttp.okhttp.OkHttpFutureBackend
+import com.softwaremill.sttp.{SttpBackend, SttpBackendOptions}
+import fr.acinq.eclair.randomBytes
+
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+
+object SttpUtils {
+
+  private lazy val backends = new SttpBackendHolder
+
+  def backend(socksProxy_opt: Option[Socks5ProxyParams]): SttpBackend[Future, Nothing] = backends.get(socksProxy_opt)
+
+  private class SttpBackendHolder {
+    @volatile
+    @transient
+    private var backends = Map.empty[Option[Socks5ProxyParams], SttpBackend[Future, Nothing]]
+
+    def get(socksProxy_opt: Option[Socks5ProxyParams]): SttpBackend[Future, Nothing] = synchronized {
```

Comment:
That's really the kind of code we want to avoid by using akka and actors (manually dealing with synchronization and memory fences). Why do you need that in the first place? Eclair will only ever have one `Socks5ProxyParams`, so why is a map needed? I'm sure we can do something much simpler...

Reply:
Maybe make it a part of `NodeParams`? It's a singleton and it has all the data to create a `SttpBackend`.